lib/index.js

```diff
@@ -46,21 +46,27 @@ exports.base64 = base64;
 exports.type = cors ? 'xhr' : 'jsonp';
 
 /**
- * Send the given `obj` to `url` with `fn(err, req)`.
+ * Send the given `obj` to `url` with `fn(err, req)`. A timeout of zero or
+ * undefined indicates no timeout.
  *
  * @param {String} url
  * @param {Object} obj
  * @param {Object} headers
+ * @param {long} timeout
  * @param {Function} fn
  * @api private
  */
 
-function json(url, obj, headers, fn) {
-  if (arguments.length === 3) fn = headers, headers = {};
-
+function json(url, obj, headers, timeout, fn) {
   var req = new XMLHttpRequest;
   req.onerror = fn;
   req.onreadystatechange = done;
+
+  if (timeout) {
```

Comment:
This code path is here so that we can skip setting a timeout for the non-retry path (i.e. leave the behaviour as is). We'll eventually want to set some timeout there as well, but today is not the day for that.